### DISCUSSION TITLE
OperationContract - IsOneWay=true

### DISCUSSION THREAD

**Discussion by JanK007** [CoreWCF/CoreWCF#1064] [Q&A]
Hello,
if I used this operation contract in WCF:

[System.ServiceModel.OperationContract(IsOneWay = true)]
void doSomething(string var1);

then the calling of this operation in .NET framework 4 has been finished immediately.
Now, I try to port this operation into .NET Core 6 + CoreWCF 1.3.2 but the calling of the operation (f.e. from SOAPUI) waits until the entire process of the operation has been finished. It seems that value of IsOneWay attribute is ignored.

Do I something wrong? Do you have some advice how to solve it without changing implementation of operation (fe. to inner thread calling)? 
Thank you.
Best regards.
Jan.

**Comment by nuzolx** [comment]
Face to the same error here.
@JanK007 did u solve it?

**Reply by mconnew** [reply]
@nuzolx, are you seeing this problem? Can you explain what you're seeing? IsOneWay isn't quite the same as fire and forget. You can still receive a Fault message back. E.g. if the specified action doesn't match a known operation, or there's a problem with deserializing the request. IsOneWay is intended to respond with a simple ACK (with HTTP that's a 200 response I think, but might be 201 accepted, with Net.Tcp it's a single byte ACK response) just before the request is dispatched. So things like message inspectors will run _before_ the ack response.

**Reply by JanK007** [reply]
Hello,
I "solved" it by invoking processing thread in method and return empty response. So I processed data asynchronously - client obtained response immediately but it's not exactly what [System.ServiceModel.OperationContract(IsOneWay = true)] means.
Regards.
Jan.

**Reply by mconnew** [reply]
Does it do the right thing if you use the attribute from the CoreWCF namespace? I wonder if this is a System.ServiceModel compatibility issue, as we do have unit tests for this.
